gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxForm,
+  NxFormGroup,
+  NxToggle,
+  useToggle,
+  NxFieldset,
+  NxRadio,
+  NxTooltip,
+  NxFontAwesomeIcon
+} from '@sonatype/react-shared-components';
+
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+const DarkModeModal = () => {
+  const [showModal, setShowModal] = useState(false);
+  const modalCloseHandler = () => setShowModal(false);
+
+  const [enableModeChange, toggleEnableModeChange] = useToggle(false);
+  const [mode, setMode] = useState<string | null>(null);
+
+  const { classList } = document.documentElement;
+
+  useEffect(() => {
+    classList.toggle('nx-html--enable-color-schemes', enableModeChange);
+  }, [enableModeChange]);
+
+  useEffect(() => {
+    classList.toggle('nx-html--dark-mode', mode === 'dark' && enableModeChange);
+    classList.toggle('nx-html--light-mode', mode === 'light' && enableModeChange);
+  }, [enableModeChange, mode]);
+
+  return (
+    <>
+      <NxButton variant="tertiary" onClick={() => setShowModal(true)}>Theme Settings</NxButton>
+      { showModal &&
+        <NxModal id="nx-modal-dark-mode-example"
+                 onCancel={modalCloseHandler}
+                 aria-labelledby="nx-modal-dark-mode-form-header">
+          <NxForm onSubmit={modalCloseHandler}
```

Comment:
The behavior and presentation around when the settings take effect is inconsistent.  The fact that the footer button is a primary button, tied to a form submission, with text "Save and Close" would seem to imply that the settings in this modal do not take effect until you click that button and close the modal.  On the other hand, the use of `NxToggle` implies an immediate change, and in fact you have the logic set up to make the changes immediately.  So we need to rectify that difference, either by making the changes not take effect until the form is submitted (and also by using `NxCheckbox` instead of `NxToggle`), or we need to make it more clear that the settings don't wait for any sort of submission, probably by removing the `NxForm` wrapper entirely and instead creating a simple `NxFooter` with a secondary or tertiary button that's just labelled "Close".

Reply:
> The behavior and presentation around when the settings take effect is inconsistent. The fact that the footer button is a primary button, tied to a form submission, with text "Save and Close" would seem to imply that the settings in this modal do not take effect until you click that button and close the modal. On the other hand, the use of `NxToggle` implies an immediate change, and in fact you have the logic set up to make the changes immediately. So we need to rectify that difference, either by making the changes not take effect until the form is submitted (and also by using `NxCheckbox` instead of `NxToggle`), or we need to make it more clear that the settings don't wait for any sort of submission, probably by removing the `NxForm` wrapper entirely and instead creating a simple `NxFooter` with a secondary or tertiary button that's just labelled "Close".

Opted to get rid of the form, which made more sense IMO - [e7e6fad](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/e7e6fad4ba850da86da7c7a127234dab4dced720)